cinn/auto_schedule/tests/single_op_program_builder.h

```diff
@@ -337,5 +337,24 @@ class LookupTableProgramBuilder : public TestProgramBuilder {
   int64_t padding_idx_;
 };
 
+class GatherProgramBuilder : public TestProgramBuilder {
+ public:
+  GatherProgramBuilder(const std::vector<int32_t>& operand_shape, const std::vector<int32_t>& index_shape, int32_t axis)
+      : operand_shape_(operand_shape), index_shape_(index_shape), axis_(axis) {}
+
+  frontend::Program operator()() override {
+    frontend::NetBuilder builder("gather_builder");
+    auto operand = builder.CreateInput(Float(32), operand_shape_, "operand");
+    auto index   = builder.CreateInput(Int(32), index_shape_, "index");
+    auto out     = builder.Gather(operand, index, axis_);
+    return builder.Build();
+  }
+
+ private:
+  std::vector<int32_t> operand_shape_;
+  std::vector<int32_t> index_shape_;
+  int32_t axis_;
+};
```

Comment:
```suggestion
class GatherProgramBuilder : public TestProgramBuilder {
 public:
  GatherProgramBuilder(const std::vector<int32_t>& operand_shape, const std::vector<int32_t>& index_shape, int32_t axis) {
      frontend::NetBuilder builder("gather_builder");
      auto operand = builder.CreateInput(Float(32), operand_shape, "operand");
      auto index   = builder.CreateInput(Int(32), index_shape, "index");
      auto out     = builder.Gather(operand, index, axis);
      program_ = builder.Build();
  }

  frontend::Program operator()() override {
    return program_;
  }

 private:
 frontend::Program program_;
};
```
把测试用例精简成这种形式是否更好一些，声明那些数据成员有点冗余

Reply:
这样确实简洁些，这个后面新提pr统一改吧，之前已经存在的单算子测试用例也改过来。